File: Week 01/id_503/leetcode_88_503.go
package week01

func merge(nums1 []int, m int, nums2 []int, n int) {

	p1, p2, p := m-1, n-1, m+n-1

	for p1 >= 0 && p2 >= 0 {

		if nums1[p1] < nums2[p2] {
			nums1[p] = nums2[p2]
			p2--
		} else {
			nums1[p] = nums1[p1]
			p1--
		}

		p--
	}

	for p2 >= 0 {
		nums1[p] = nums2[p2]
		p--
		p2--
	}
}

File: Week 01/id_503/leetcode_88_503.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function (nums1, m, nums2, n) {
    let p1 = m - 1;
    let p2 = n - 1;
    let p = m + n - 1;
    while (p1 >= 0 && p2 >= 0) {
        nums1[p--] = nums1[p1] < nums2[p2] ? nums2[p2--] : nums1[p1--];
    }

    while (p2 >= 0) {
        nums1[p--] = nums2[p2--]
    }
};